activesupport/test/current_attributes_test.rb

```diff
@@ -191,6 +191,7 @@ def after_teardown
     end
     assert_equal 42, enumerator.next
   ensure
+    Session.current = nil
```

Comment:
This still leaks an ActiveSupport::CurrentAttributes instance in the thread local variables. I hadn't seen that and I had started a branche that was using clear_all: https://github.com/etiennebarrie/rails/commit/ec37c240d03d8200647ce060b2f27a4924eee0a6 I think that would be preferable.

Reply:
👍

Reply:
@etiennebarrie I'm ok with merging your branch as well.

Reply:
Hum, could we call `clear_all` when `_use_thread_variables` changes? I mean in the implementation itself, not just tests.

Also @etiennebarrie please open a PR.

Reply:
Yeah good catch to put it in the config change, that way we never leave objects behind.

Also I wonder if we shouldn't use `clear_all` more often (e.g. instead of `reset_all`): I'm not sure keeping the object around saves a lot of allocations (usually it would be a few current attributes instances per request only).

Or if the idea is that they should be like singleton objects, staying alive forever once they're allocated, but I doubt people hold onto the instance:

```ruby
class Foo < ActiveSupport::CurrentAttributes
  attribute :foo
end

# then
object = Foo.instance
# later
object.foo
```

In which case keeping the same object would matter.